File: 2021/vika5/egill/walrus2.cpp
#include <iostream>

using namespace std;

#define goal 1000
#define limit 2001

int main()
{
    int n, i, j;
    cin >> n;
    int w[n];
    for(i = 0; i < n; i++) {
        cin >> w[i];
    }

    bool check[limit] = {false};
    check[0] = 1;

    i = 0;
    while (i < n) {
        int curr = w[i];
        int top = limit - curr;

        for (j = top; j >= 0; j--) {
            if (check[j]) check[j+curr] = 1;
        }
        i++;
    }



    for (i = 0; i < limit; i++) {
        if (check[goal + i]) {
            int output = goal+i;
            cout << output << '\n';
            break;
        } else if (check[goal-i]) {
            int output = goal-i;
            cout << output << '\n';
            break;
        }
    }

    return 0;
}

File: 2021/vika05/walrus2.py

n = int(input())
w = [int(input()) for i in range(n)]

print( len(w) )

goal = 1000
limit = 2001

check = [0 for i in range(limit)]
check[0] = 1

i = 0
while i < n:
    curr = w[i]
    top = limit - curr

    for j in range(top,-1,-1):
        if check[j]:
            check[j+curr] = 1
    i+=1

for i in range(limit):
    if check[goal+i]:
        print(goal+i)
        break
    else:
